cli/js/util.ts

```diff
@@ -78,3 +79,12 @@ export function immutableDefine(
     writable: false,
   });
 }
+
+// @internal
+export function pathFromURL(url: URL): string {
+  return build.os == "windows"
+    ? new URL(url).pathname
+        .replace(/^\/*([A-Za-z]:)(\/|$)/, "$1/")
+        .replace(/\//g, "\\")
+    : new URL(url).pathname;
+}
```

Comment:
I would have expected something like 

```ts
if (url.schema == "file") {
  return url.pathname
} else {
  throw TypeError("Not a file URL");
}
```
This seems to be an important function - we need to figure out a way to have unit tests for it. Unfortunately, we only have a testing mechanism for public exposed methods. I think we should make use of "internalSymbol" to try to export this secretly from the Deno namespace for testing...

Reply:
@ry https://github.com/denoland/deno/blob/f0ba814a79fe4f0791c4b0ef496a704356597467/cli/js/internals.ts#L8-L16

Reply:
@ry Also, this function is pieced together from `std`: https://github.com/denoland/deno/blob/f0ba814a79fe4f0791c4b0ef496a704356597467/std/path/posix.ts#L437-L439 https://github.com/denoland/deno/blob/f0ba814a79fe4f0791c4b0ef496a704356597467/std/path/win32.ts#L916-L920 so as long as _they're_ tested and this one is kept in sync... 🤷‍♂️